cache.go

```diff
@@ -169,12 +169,13 @@ func (cache *Cache) SetWithTTL(key string, data interface{}, ttl time.Duration)
 func (cache *Cache) Get(key string) (interface{}, bool) {
 	item, exists := cache.getItem(key)
 	if exists {
+		cache.mutex.RLock()
```

Comment:
another potential solution would be instead of converting this to an RWMutex and RLocking here, this function could simply `return cache.getItem(key)` - but I guess that could potentially change behavior if the item is ever non-nil when exists is false?